Sources/SwiftParserDiagnostics/ParserDiagnosticMessages.swift

```diff
@@ -386,6 +386,14 @@ public struct MissingAttributeArgument: ParserError {
   }
 }
 
+public struct MissingBothStringQuotesOfStringSegments: ParserError {
+  public let stringSegments: StringLiteralSegmentsSyntax
```

Comment:
The variable name is `stringSegments` but is not an array?
Is it?

Reply:
This is not an array, but I declared it this way to align the node's property names and meanings passed when generating messages. Examples include `components` in `TrailingVersionAreIgnored` or `unexpectedNodes` in `UnexpectedNodesError`. 
Do you think it would be better to declare it simply as `stringSegment`?

Reply:
After diving in the code, it looks like `StringLiteralSegmentsSyntax` also is a `SyntaxCollection`. Just like `TrailingVersionAreIgnored ` and `UnexpectedNodesError `
So you can completely ignore this comment 🫠

Reply:
Ah, it seems that all of these have a common protocol called `SyntaxCollection`! Thank you for your feedback 🙇‍♂️